core/src/main/java/org/infinispan/affinity/impl/KeyAffinityServiceImpl.java

```diff
@@ -131,12 +129,17 @@ public K getKeyForAddress(Address address) {
             } finally {
                maxNumberInvariant.readLock().unlock();
             }
+            try{
+                Thread.currentThread().sleep(POOL_INTERVAL);
```

Comment:
I think we could poll the queue here instead of waiting for a fixed amount of time: `queue.poll(POOL_INTERVAL, TimeUnit.MILLISECONDS);`